### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by mmmries**
_Note: I'm not ready to publish any versions of this yet so I'm just going to be doing some small PRs to get it ready_

This updates the dependencies in the gemspec to avoid picking up major version bumps in `bunny` or `march_hare` since those might break an API that we depend on. I've also updated the `Configuration` class to have the idempotent loading we use in ActionSubscriber and only keep configurations for things that `ActivePublisher` cares about (ie. no `prefetch` etc).

/cc @film42

### CODE REVIEW

File: lib/active_publisher/configuration.rb [hunk 1]
```diff
@@ -1,23 +1,86 @@
+require "yaml"
+
 module ActivePublisher
   class Configuration
-    attr_accessor :default_exchange,
-                  :error_handler,
+    attr_accessor :async_publisher,
```
Comment (liveh2o) [inline lib/active_publisher/configuration.rb:5]: Alphabetize?

File: lib/active_publisher/configuration.rb [hunk 2]
```diff
@@ -1,23 +1,86 @@
...
+      :timeout => 1,
+      :username => "guest",
+      :password => "guest",
+      :virtual_host => "/"
+    }
+
+    ##
+    # Class Methods
+    #
+    def self.configure_from_yaml_and_cli(cli_options = {}, reload = false)
```
Comment (liveh2o) [inline lib/active_publisher/configuration.rb:43]: This method is pretty large. Perhaps we could take this opportunity to split it up?

File: lib/active_publisher/configuration.rb [hunk 3]
```diff
@@ -1,23 +1,86 @@
...
-                  :timeout
+                  :seconds_to_wait_for_graceful_shutdown,
+                  :username,
+                  :timeout,
+                  :virtual_host
+
+    CONFIGURATION_MUTEX = ::Mutex.new
+
+    DEFAULTS = {
+      :async_publisher => 'memory',
```
Comment (liveh2o) [inline lib/active_publisher/configuration.rb:24]: Mixed quotes.

**Comment by mmmries** [comment]
Woops, I got a :ship: on #2, but forgot that it included these changes which still had some outstanding comments. I'll open another PR with the suggested changes from @liveh2o

**Comment by film42** [comment]
:eggplant:

**Pull request merged**